android/src/main/java/com/rudderstack/android/internal/infrastructure/ReinstatePlugin.kt

```diff
@@ -153,6 +144,7 @@ class ReinstatePlugin : Plugin {
             if (shouldMigrateContext()) {
                 _analytics?.migrateContextFromV1()
             }
```

Comment:
If current context is created already, and then we replace the same with older context, this will mean events will switch back to old context. 
This is a known issue and have a separate ticket for the same

Reply:
Known issue after fix for events in between
[issue](https://linear.app/rudderstack/issue/SDK-1579/edge-cases-on-migration-from-v1-to-v2known-issues)